base/compiler/tfuncs.jl

```diff
@@ -466,7 +466,7 @@ add_tfunc(<:, 2, 2,
               return Bool
           end, 0)
 
-function const_datatype_getfield_tfunc(@nospecialize(sv), @nospecialize(fld))
+function const_datatype_getfield_tfunc(@nospecialize(sv), fld::Int)
```

Comment:
That seems like a good thought, but only one of the callers actually ensures that it passes an Int here

Reply:
Good catch, I should add a check for that.